apis/core/types_shared.go

```diff
@@ -106,6 +106,8 @@ const (
 	ErrorCleanupResources ErrorCode = "ERR_CLEANUP"
 	// ErrorConfigurationProblem indicates that the last error occurred due a configuration problem.
 	ErrorConfigurationProblem ErrorCode = "ERR_CONFIGURATION_PROBLEM"
+	// ErrorTimeout indicates that an operation timed out.
+	ErrorTimeout ErrorCode = "ERR_TIMEOUT"
```

Comment:
I think it can be 'unretryable', but it also can be resolved by retrying, depending on the situation. E.g. if the deployer is currently unavailable (due to node rolling, ...) retrying at a later time could help. If the deployer isn't there, however, no amount of retries will solve the problem.
For other timeouts, such as the progressing timeout for example, retrying might have an even higher chance of solving the problem.
I think we should leave it retryable for now ... WDYT?

Reply:
You're basically right. But currently we have the issue that the behavior for installations that were in failed state and switch to successfull without being triggered is undefined.

So I would make it not retryable and then later when we have proper propagation for such scenarios we will remove it again and make it retryable.